include/villas/nodes/delta_sharing/Protocol.h

```diff
@@ -0,0 +1,298 @@
+/* Node type: Delta Share.
+ *
+ * Author: Ritesh Karki <ritesh.karki@rwth-aachen.de>
+ * SPDX-FileCopyrightText: 2014-2023 Institute for Automation of Complex Power Systems, RWTH Aachen University
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <iostream>
+#include <villas/nodes/delta_sharing/jansson_wrapper.h>
+#include <map>
+#include <string>
+#include <vector>
+#include <boost/optional.hpp>
+#include <boost/optional/optional_io.hpp>
+
+namespace DeltaSharing
```

Comment:
Please format this code with clang-format. There is also a script to do so in the `tools` folder, or you can use pre-commit.